components/Swap/Withdraw/Wallet/ZKsyncWalletWithdraw.tsx

```diff
@@ -0,0 +1,114 @@
+import { useSettingsState } from '../../../../context/settings';
+import { Network } from 'zksync/build/types';
+import KnownInternalNames from '../../../../lib/knownIds';
+import { useNetwork } from 'wagmi';
+
+type Props = {
+    depositAddress: string,
+    amount: number
+}
+
+const ZkSyncWalletWithdrawStep: FC<Props> = ({ depositAddress, amount }) => {
+    const [loading, setLoading] = useState(false);
+    const [transferDone, setTransferDone] = useState<boolean>();
+    const { setZkSyncAccount } = useWalletUpdate();
+    const { setSwapTransaction } = useSwapTransactionStore();
+    const { swap } = useSwapDataState();
+    const [syncWallet, setSyncWallet] = useState<zksync.Wallet>(null);
+    const signer = useEthersSigner();
+    const { chain } = useNetwork();
+    
+    const { networks } = useSettingsState();
+    const { source_network: source_network_internal_name } = swap;
+    const source_network = networks.find(n => n.internal_name === source_network_internal_name);
+    const source_currency = source_network.currencies.find(c => c.asset.toLocaleUpperCase() === swap.source_network_asset.toLocaleUpperCase());
+    const defaultProvider = swap?.source_network?.split('_')?.[1]?.toLowerCase() == "mainnet" ? "mainnet" : "georli";
+    const ethereum = networks.find(n => n.internal_name === KnownInternalNames.Networks.EthereumMainnet);
+
+    const handleConnect = async () => {
+        setLoading(true)
+        try {
+            const syncProvider = await zksync.getDefaultProvider(defaultProvider as Network);
+            const wallet = await zksync.Wallet.fromEthSigner(signer, syncProvider);
+            wallet.getAccountState()
+            setSyncWallet(wallet);
+            setZkSyncAccount(wallet.cachedAddress);
+        }
+        catch (e) {
+            toast(e.message)
+        }
+        setLoading(false)
+    }
+
+    const handleTransfer = async () => {
+        setLoading(true)
+        try {
+            const tf = await syncWallet.syncTransfer({
+                to: depositAddress,
+                token: swap?.source_network_asset,
+                amount: zksync.closestPackableTransactionAmount(utils.parseUnits(amount.toString(), source_currency?.decimals)),
+                validUntil: zksync.utils.MAX_TIMESTAMP - swap?.sequence_number,
+            });
+
+            const res = await tf.awaitReceipt();
+            if (res?.success) {
+                setSwapTransaction(swap?.id, PublishedSwapTransactionStatus.Completed, tf?.txHash?.replace('sync-tx:', ''));
+                setTransferDone(true)
+            } else {
+                setSwapTransaction(swap?.id, PublishedSwapTransactionStatus.Error, tf?.txHash?.replace('sync-tx:', ''), res?.failReason);
+            }
+        }
+        catch (e) {
+            if (e?.message)
+                toast(e.message)
+        }
+        setLoading(false)
+    }
+
+    if (!signer) {
```

Comment:
Here you should check if there's a syncWallet, and set it to null or with something else, because:

To reproduce
1. Connect a wallet
2. Click Unlock zkSync account
3. Click disconnect account (x)
4. Connect a different wallet
5. It doesn't show Unlock zkSync account, because the sync wallet is not null

Maybe you should store not a single syncWallet instance in the state, but a dictionary [address, syncWallet]

Reply:
Another option:
Maybe instead of storing syncWallet here and a zksyncAccount in wallet context, you can store syncWallet in wallet context

Reply:
That way, when disconnecting, syncWallet will also be gone